src/pages/iou/request/step/IOURequestStepAmount.tsx

```diff
@@ -117,7 +147,57 @@ function IOURequestStepAmount({
         // If a reportID exists in the report object, it's because the user started this flow from using the + button in the composer
         // inside a report. In this case, the participants can be automatically assigned from the report and the user can skip the participants step and go straight
         // to the confirm step.
-        if (report?.reportID) {
+        if (report?.reportID && !ReportUtils.isArchivedRoom(report)) {
+            const selectedParticipants = IOU.setMoneyRequestParticipantsFromReport(transactionID, report);
+            const participants = selectedParticipants.map((participant) => {
+                const participantAccountID = participant?.accountID ?? 0;
+                return participantAccountID ? OptionsListUtils.getParticipantsOption(participant, personalDetails) : OptionsListUtils.getReportOption(participant);
+            });
+            const backendAmount = CurrencyUtils.convertToBackendAmount(Number.parseFloat(amount));
+
+            if (shouldSkipConfirmation) {
+                if (iouType === CONST.IOU.TYPE.SPLIT) {
+                    IOU.splitBillAndOpenReport({
+                        participants,
+                        currentUserLogin: currentUserPersonalDetails.login ?? '',
+                        currentUserAccountID: currentUserPersonalDetails.accountID ?? 0,
+                        amount: backendAmount,
+                        comment: '',
+                        currency,
+                        merchant: '',
+                        tag: '',
+                        category: '',
+                        created: transaction?.created ?? '',
+                        billable: false,
+                        iouRequestType: CONST.IOU.REQUEST_TYPE.MANUAL,
+                    });
+                    return;
+                }
+                if (iouType === CONST.IOU.TYPE.SEND) {
+                    if (paymentMethod && paymentMethod === CONST.IOU.PAYMENT_TYPE.EXPENSIFY) {
+                        IOU.sendMoneyWithWallet(report, backendAmount, currency, '', currentUserPersonalDetails.accountID, participants[0]);
+                        return;
+                    }
```

Comment:
We still need to playSound if we submit/send money via QAB.

More context: https://github.com/Expensify/App/issues/51520#issuecomment-2439718525